lib/Gocdb_Services/Site.php

```diff
@@ -1429,39 +1429,74 @@ public function deleteAPIAuthEntity(\APIAuthentication $authEntity, \User $user)
         $authEntServ->deleteAPIAuthentication($authEntity);
     }
 
-    public function editAPIAuthEntity(\APIAuthentication $authEntity, \User $user, $newValues) {
-
+    /**
+     * Helper to edit an `APIAuthentication` entity.
+     *
+     * @param \APIAuthentication $authEntity `APIAuthentication` entity to check.
+     * @param \User $user User doing the edit.
+     * @param mixed $newValues Holds the new data for updating the `APIAuthentication` entity.
+     */
+    public function editAPIAuthEntity(\APIAuthentication $authEntity, \User $user, $newValues)
+    {
         $parentSite = $authEntity->getParentSite();
 
         // Check the user can do this. Thows exception if not.
         $this->checkUserAuthz($user, $parentSite);
 
-        $identifier = $newValues['IDENTIFIER'];
-        $type = $newValues['TYPE'];
-
         /**
         @var org\gocdb\services\APIAuthenticationService
         */
         $authEntServ = \Factory::getAPIAuthenticationService();
         $authEntServ->setEntityManager($this->em);
 
+        if (!($newValues['isRenewalRequest'])) {
+            $this->validOIDCSubjectIdentifier($authEntity, $newValues, $parentSite, $authEntServ);
+        }
+
+        $authEntServ->editAPIAuthentication($authEntity, $user, $newValues);
+
+        return $authEntity;
+    }
+
+    /**
+     * Helper to check if the given identifier is valid for an OIDC Subject.
+     *
+     * @param \APIAuthentication $authEntity `APIAuthentication` entity to check.
+     * @param mixed $newValues Holds the new data for updating the `APIAuthentication` entities.
+     * @param \Site $parentSite Site details to validate whether the given identifier is unique.
+     * @param \APIAuthenticationService $authEntServ Singleton `APIAuthenticationService` service.
+     *
+     * @throws \Exception If the given identifier is invalid OIDC Subject.
+     */
+    private function validOIDCSubjectIdentifier(
```

Comment:
I think this method will check the validate for all identifiers, not limited to OIDC subjects (it just does a but more detailed checking on OIDC subjects).

maybe something like

```suggestion
    /**
     * Helper to check if the given identifier is valid,
     * possibly applying type specific checks.
     *
     * @param \APIAuthentication $authEntity `APIAuthentication` entity to check.
     * @param mixed $newValues Holds the new data for updating the `APIAuthentication` entities.
     * @param \Site $parentSite Site details to validate whether the given identifier is unique.
     * @param \APIAuthenticationService $authEntServ Singleton `APIAuthenticationService` service.
     *
     * @throws \Exception If the given identifier is invalid.
     */
    private function validIdentifier(
```